### DISCUSSION TITLE
Errors while installing SellersApi and

### DISCUSSION THREAD

**Discussion by iololife** [amzn/selling-partner-api-models#3507] [General] [answered]
I'm running into two issues while following the steps outlined at https://developer-docs.amazon.com/sp-api-blog/docs/automate-your-sp-api-calls-using-java-sdk and would appreciate any help on these issues.

1. In Sellers_JavaCL, when I run "mvn package" from api directories as instructed in READ.md, I get "cannot find symbol" unless I add a dependency for sellingpartnerapi-aa-java to pom.xml: 
`<dependency>
            <groupId>com.amazon.sellingpartnerapi</groupId>
            <artifactId>sellingpartnerapi-aa-java</artifactId>
            <version>1.0</version>
 </dependency>`

However, fixing this way leads to

2. I've followed through all the steps outlined at https://developer-docs.amazon.com/sp-api-blog/docs/automate-your-sp-api-calls-using-java-sdk and I was able to authenticate and get a valid response back using the Builder from io.swagger.client.api.SellersApi. I then attempted to use io.swagger.client.api.ListingsApi and I got "cannot resolve symbol" error. After adding the dependency `<dependency>
            <groupId>com.amazon.sellingpartnerapi</groupId>
            <artifactId>sellingpartnerapi-aa-java</artifactId>
            <version>1.0</version>
 </dependency>` to pom.xml and running "mvn package" followed by "mvn clean install" from the api directory for ListingsApi, I was able to use ListingsApi, but that caused SellersApi to come up as "symbol not found" error in my app. What do I need to do to be able to use two or more api's form io.swagger.client.api.* in my app?

Thanks,
Kevin

**Comment by sungolivia** [comment]
@puppsupr can you take a look?

**Reply by puppsupr** [reply]
Hi @iololife,
Can you please provide me the screenshots for the same ?

Thanks and Regards,
Supriya P
Solutions Architect, Selling Partner API
[SP API Blogs](https://developer-docs.amazon.com/sp-api-blog/docs/getting-started) | [SP API YouTube Channel](https://www.youtube.com/@amazon-sp-api)

**Comment by iololife** [comment]
Originally, SELLERS_JAVACL/pom.xml doesn't have
`<dependency>
            <groupId>com.amazon.sellingpartnerapi</groupId>
            <artifactId>sellingpartnerapi-aa-java</artifactId>
            <version>1.0</version>
 </dependency>`
Without  this dependency in SELLERS_JAVACL/pom.xml, when I run "mvn package" or "mvn clean install", I get
<img width="1371" alt="Screenshot 2023-02-14 at 3 33 56 PM" src="https://user-images.githubusercontent.com/94832286/218886916-a006376f-66c8-486f-8530-89c256531256.png">

If I add the dependency for sellingpartnerapi-aa-java, "mvn clean install" succeeds.
<img width="880" alt="Screenshot 2023-02-14 at 3 44 18 PM" src="https://user-images.githubusercontent.com/94832286/218888131-f6225736-22d2-482c-973b-79e8a6761099.png">

I can then run the example code below which returns the expected result.

SellersApi sellersApi = new SellersApi.Builder()
                .awsAuthenticationCredentials(awsAuthenticationCredentials)
                .lwaAuthorizationCredentials(lwaAuthorizationCredentials)
                .awsAuthenticationCredentialsProvider(awsAuthenticationCredentialsProvider)
                .endpoint("https://sellingpartnerapi-na.amazon.com")
                .build();
try {
        GetMarketplaceParticipationsResponse result = sellersApi.getMarketplaceParticipations();
        System.out.println(result);
} catch (ApiException e) {
        System.err.println("Exception when calling SellersApi#getMarketplaceParticipations");
        e.printStackTrace();
}

My main issue is that when I solve this way, I can only use either SellersApi or ListingsApi, but not both

<img width="727" alt="Screenshot 2023-02-14 at 7 57 00 PM" src="https://user-images.githubusercontent.com/94832286/218925504-6c8d98cd-f692-4c13-9e6c-a0e00707f152.png">
or 
<img width="740" alt="Screenshot 2023-02-14 at 7 58 24 PM" src="https://user-images.githubusercontent.com/94832286/218925730-7624efc6-3d29-40f5-8e8f-cd624759417a.png">

**Comment by iololife** [comment, marked as answer]
Never mind, I got this to work. Didn't realize that the pom.xml in SwaggerToCL was for all modules.

**Reply by bbq572** [reply]
@iololife -- I ran into this as well and found you posts about it, I asked support on how to properly configure the pom.xmls to allow use of multiple modules -- no responses -- can you reply what you did to get past this issue  -- not sure what I missed -- THANKS!
